src/uberjob/progress/_progress.py

```diff
@@ -19,11 +19,15 @@
 
 
 class Progress:
-    """A way to observe progress."""
+    """
+    A way to observe progress.
 
-    def __init__(self, observer: Callable[[], ProgressObserver]):
-        self._observer = observer
+    :param create_observer: A callable that creates a single-use progress observer.
+    """
+
+    def __init__(self, create_observer: Callable[[], ProgressObserver]):
```

Comment:
I renamed the parameter from "observer" to "create_observer" so the docs would read more clearly. I left the method as "observer" though.